symengine/pow.cpp

```diff
@@ -311,29 +323,52 @@ RCP<const Basic> Log::create(const RCP<const Basic> &a) const
 
 RCP<const Basic> log(const RCP<const Basic> &arg)
 {
-    if (eq(*arg, *zero)) {
-        throw NotImplementedError(
-            "log(0) is complex infinity. Yet to be implemented");
-    }
+    if (eq(*arg, *zero))
+        return ComplexInf;
     if (eq(*arg, *one))
         return zero;
     if (eq(*arg, *E))
         return one;
+    if (eq(*arg, *Inf))
```

Comment:
Yes, you need such a class, but it doesn't need to be templated.
Sorry about the confusion, I thought this was about `pow` since it is in the `pow.cpp` file, but it's about `log`.